src/server.ts

```diff
@@ -4,31 +4,221 @@ import express from "express";
+  }
+});
+
+//-------------------------------------------------------------------Get all users
+app.get("/users", async (req, res) => {
+  try {
+    const query = "SELECT * FROM users";
+    const response = await client.query(query);
+    res.status(200).send(response.rows);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+//-------------------------------------------------------------------Get all likes for a specific post
+app.get("/resources/:resourceID/likes", async (req, res) => {
+  const resourceID = req.params.resourceID;
+  try {
+    const query =
+      "SELECT count (*) FROM likes WHERE resource_id = $1 AND is_liked IS NOT NULL GROUP BY is_liked";
+    const values = [resourceID];
+    const response = await client.query(query, values);
+    res.status(200).send(response.rows);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+//-------------------------------------------------------------------Get whether the signed in user liked the resource
+app.get("/resources/:resourceID/likes/:userID", async (req, res) => {
+  const { userID, resourceID } = req.params;
+  try {
+    const query =
+      "SELECT is_liked FROM likes WHERE resource_id = $1 AND user_id = $2";
+    const values = [resourceID, userID];
+    const response = await client.query(query, values);
+    res.status(200).send(response.rows);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+//-------------------------------------------------------------------Get all comments on a post
+app.get("/resources/:resourceID/comments", async (req, res) => {
+  const { resourceID } = req.params;
+  try {
+    const query = "SELECT * FROM comments WHERE resource_id = $1";
+    const values = [resourceID];
+    const response = await client.query(query, values);
+    res.status(200).send(response.rows);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+//-------------------------------------------------------------------Get todo items for a user
+app.get("/to-do-list/:userID", async (req, res) => {
+  const { userID } = req.params;
+  try {
+    const query =
+      "SELECT * FROM to_do_list join resources on to_do_list.resource_id = resources.resource_id and to_do_list.user_id = $1";
+    const values = [userID];
+    const response = await client.query(query, values);
+    res.status(200).send(response.rows);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+//-------------------------------------------------------------------POST REQUESTS
+//-------------------------------------------------------------------Post resource to database
+app.post("/resources", async (req, res) => {
+  const resource = req.body;
+  // const tagsArray = `ARRAY[${resource.tags}]`
+  try {
+    const query =
+      "INSERT INTO resources (resource_url, author_name, resource_name, " +
+      "resource_description, tags, content_type, selene_week, usage_status, recommendation_reason, user_id) " +
+      "VALUES ($1,$2,$3,$4, ARRAY[$5],$6,$7,$8,$9,$10)";
```

Comment:
Just wanted to flag your use of an Array type here - Neill has sent me lots of db resources over the weekend concerning normalization - essentially, using arrays in db's in not usually preferred (apparently) so you may want to consider the use of a separate table for tags that utilises a many-to-many relationship between resources and tags. I'm still learning this myself, but it might be worth the research 👍